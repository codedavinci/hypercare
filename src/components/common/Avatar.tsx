export type AvatarProps = {
  src: string;
  altText?: string;
};

export default function Avatar({ src, altText = " " }: AvatarProps) {
  return (
    <div className="avatar">
      <div className="w-24 rounded-full ring ring-primary ring-offset-base-100 ring-offset-2">
        <img src={src} alt={altText} />
      </div>
    </div>
  );
}

Avatar.displayName = "Avatar";
