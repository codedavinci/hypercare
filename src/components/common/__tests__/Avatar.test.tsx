import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import Avatar from "../Avatar";

describe("Avatar", () => {
  it("renders correctly and matches snapshot", () => {
    const { asFragment } = render(
      <Avatar src="https://via.placeholder.com/150" altText="User Avatar" />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders with correct src and alt attributes", () => {
    const src = "https://via.placeholder.com/150";
    const altText = "User Avatar";
    render(<Avatar src={src} altText={altText} />);

    const imgElement = screen.getByAltText(altText);
    expect(imgElement).toHaveAttribute("src", src);
    expect(imgElement).toHaveAttribute("alt", altText);
  });

  // Unit test for default alt attribute
  it("renders with default alt text if none is provided", () => {
    const src = "https://via.placeholder.com/150";
    render(<Avatar src={src} />);

    const imgElement = screen.getByRole("img");
    expect(imgElement).toHaveAttribute("src", src);
    expect(imgElement).toHaveAttribute("alt", " ");
  });
});
