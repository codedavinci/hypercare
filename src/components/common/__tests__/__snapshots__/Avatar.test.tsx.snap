// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar renders correctly and matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="avatar"
  >
    <div
      class="w-24 rounded-full ring ring-primary ring-offset-base-100 ring-offset-2"
    >
      <img
        alt="User Avatar"
        src="https://via.placeholder.com/150"
      />
    </div>
  </div>
</DocumentFragment>
`;
